Implement canary deployments (#54)
gruntwork-io/helm-kubernetes-services: charts/k8s-service/README.md

- `Deployment`: The main `Deployment` controller that will manage the application container image specified in the
                `containerImage` input value.

> - `CanaryDeployment` An optional `Deployment` controller that will manage a [canary deployment](https://martinfowler.com/bliki/CanaryRelease.html) of the application container image specified in the `canary.containerImage` input value. This is useful for testing a new application tag, in parallel to your stable tag, prior to rolling the new tag out. Created only if you configure the `canary.containerImage` values (and set `canary.enabled = true`). 

Reviewer: After rereading this, just realized that this should probably be:

```suggestion
- Secondary `Deployment` for use as canary: An optional `Deployment` controller that will manage a [canary deployment](https://martinfowler.com/bliki/CanaryRelease.html) of the application container image specified in the `canary.containerImage` input value. This is useful for testing a new application tag, in parallel to your stable tag, prior to rolling the new tag out. Created only if you configure the `canary.containerImage` values (and set `canary.enabled = true`). 
```

`applicationName` will cause the `Deployment` resource to be deleted, and then created. This can lead to down time
because the resources are replaced in an uncontrolled fashion.

## How do I create a canary deployment? 

You may optionally configure a [canary deployment](https://martinfowler.com/bliki/CanaryRelease.html) of an arbitrary tag that will run as an individual deployment behind your configured service. This is useful for ensuring a new application tag runs without issues prior to fully rolling it out. 

To configure a canary deployment, set `canary.enabled = true` and define the `containerImage` values. Typically, you will want to specify the tag of your next release candidate:

```yaml
canary:
  enabled: true
    containerImage:
      repository: nginx
      tag: 1.15.9 
```
Once deployed, your service will route traffic across both your stable and canary deployments, allowing you to monitor for and catch any issues early.

## How do I verify my canary deployment? 

> Canary deployment pods have the same name as your stable deployment pods, with the addition "-canary" appended to the end, like so: 

Reviewer: ```suggestion
Canary deployment pods have the same name as your stable deployment pods, with the addition `-canary` appended to the end, like so: 
```

Reviewer: Super seded by Josh's suggestion down below.

Author: Fixed in 5957d07

> Canary deployment pods have the same name as your stable deployment pods, with the addition "-canary" appended to the end, like so: 

Reviewer 2: ```suggestion
Canary deployment pods have the same name as your stable deployment pods, with the additional `-canary` appended to the end, like so: 
```

```
Once deployed, your service will route traffic across both your stable and canary deployments, allowing you to monitor for and catch any issues early.

> ## How do I verify my canary deployment? 

Author: I need to verify these commands / output once I have my code changes implemented and running locally / in my GKE staging environment - but adding them here for now.

Author: Confirmed locally - these docs are correct and the rollback procedure works.
